chain/client/src/sync/state/network.rs

```diff
@@ -69,18 +69,21 @@ impl StateSyncDownloadSourcePeerSharedState {
                 None => PartIdOrHeader::Header,
             },
         };
-        if let Some(request) = self.pending_requests.get(&key) {
-            if request
-                .peer_id
-                .as_ref()
-                .is_some_and(|expecting_peer_id| expecting_peer_id == &peer_id)
-            {
-                let value = self.pending_requests.remove(&key).unwrap();
-                let _ = value.sender.send(data);
-                return Ok(());
-            }
+
+        let Some(request) = self.pending_requests.get(&key) else {
+            tracing::debug!(target: "sync", "Received {:?} expecting {:?}", key, self.pending_requests.keys());
```

Comment:
Think it should be OK to debug log all the pending keys since the limit is fairly low (10).